rir/src/utils/configurations.cpp

```diff
@@ -51,6 +51,7 @@ void Configurations::defaultOptimizations() {
         optimizations.push_back(new pir::Constantfold());
         optimizations.push_back(new pir::Cleanup());
         optimizations.push_back(new pir::DelayInstr());
+        optimizations.push_back(new pir::HoistInstruction());
```

Comment:
Maybe it is time to change the name of DelayInstr and also maybe be more specific with HoistInstruction? For instance, `DelayLookup`, `HoistEffectless`?

Reply:
it's not HoistEffectless. yes I agree that we should review the passes, also check if they actually work, remove stupid ones, cleanup naming and so on. but not really related to this pr